### PR TITLE
fix: add `maxDuration` to `PageConfig` type

### LINKED ISSUE #55957: `maxDuration` is missing in `PageConfig`
### Link to the code that reproduces this issue

https://github.com/vercel/next.js/blob/ca57258dbd4d5b3aba119088e2c188bf0f4671bc/packages/next/types/index.d.ts#L113-L152

### To Reproduce

1. In a page, add

```ts
import { PageConfig } from 'next'

export const config: PageConfig = {
  maxDuration: 60
}
```

### Current vs. Expected behavior

Current: Type error because `maxDuration` is not a valid property.
Expected: it is typed according to https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
irrelevant
```

### Which area(s) are affected? (Select all that apply)

TypeScript (plugin, built-in types)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by simonhaenisch**
Relevant docs: https://vercel.com/docs/functions/serverless-functions/runtimes#max-duration

Got an "action required" email about this, updated `next`, still got a type error on my `config: PageConfig` export.

The JSDoc description has been copied from the docs (slightly adjusted).

Fixes #55957.

### CODE REVIEW

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 617e539f0b678e8a2a65c0c67a441584abf42365

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 617e539f0b678e8a2a65c0c67a441584abf42365

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

File: packages/next/types/index.d.ts [hunk 1]
```diff
@@ -136,6 +136,11 @@ export type PageConfig = {
     externalResolver?: true
   }
   env?: Array<string>
+  /**
+   * Configures the longest time a serverless function can process an HTTP request
+   * before responding.
```
Comment (simonhaenisch) [inline packages/next/types/index.d.ts:141]:
```suggestion
   * Configures the longest time in seconds a serverless function can process an HTTP
   * request before responding.
```

**Comment by simonhaenisch** [comment]
@ijjk it's been like 6 weeks with no attention for this PR, could you maybe at least approve this to run in the CI? It's only adding a single missing property to an interface.

**Comment by edwinbernadus** [comment]
need this PR
webstorm IDE show many errors from this condition

**Review by huozhi — CHANGES_REQUESTED** [review]
`maxDuration` is on top-level export, it's not part of the page config. We've already supported the type checking for this in nextjs typescript plugin

**Comment by simonhaenisch** [comment]
hm weird apparently i didn't read properly (unless this was changed in the meantime 🤷🏻‍♂️)

**Review by huozhi — APPROVED** [review]
Yeah right we need it for pages as well, thanks!

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong></summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| buildDuration | 10.7s | 10.7s | N/A |
| buildDurationCached | 6s | 6.7s | ⚠️ +674ms |
| nodeModulesSize | 199 MB | 199 MB | ⚠️ +306 B |
| nextStartRea..uration (ms) | 421ms | 423ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| 199-HASH.js gzip | 30.7 kB | 30.7 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.3 kB | 53.3 kB | ✓ |
| 494.HASH.js gzip | 180 B | 181 B | N/A |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 240 B | N/A |
| main-HASH.js gzip | 31.7 kB | 31.7 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 100 kB | 100 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| _app-HASH.js gzip | 194 B | 195 B | N/A |
| _error-HASH.js gzip | 182 B | 181 B | N/A |
| amp-HASH.js gzip | 501 B | 503 B | N/A |
| css-HASH.js gzip | 322 B | 323 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.5 kB | ✓ |
| edge-ssr-HASH.js gzip | 253 B | 255 B | N/A |
| head-HASH.js gzip | 348 B | 347 B | N/A |
| hooks-HASH.js gzip | 369 B | 368 B | N/A |
| image-HASH.js gzip | 4.27 kB | 4.27 kB | N/A |
| index-HASH.js gzip | 256 B | 256 B | ✓ |
| link-HASH.js gzip | 2.61 kB | 2.6 kB | N/A |
| routerDirect..HASH.js gzip | 311 B | 311 B | ✓ |
| script-HASH.js gzip | 384 B | 383 B | N/A |
| withRouter-HASH.js gzip | 307 B | 308 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 3.17 kB | 3.17 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| _buildManifest.js gzip | 484 B | 483 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 538 B | 541 B | N/A |
| withRouter.html gzip | 523 B | 523 B | ✓ |
| Overall change | 1.05 kB | 1.05 kB | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| edge-ssr.js gzip | 92.6 kB | 92.6 kB | N/A |
| page.js gzip | 145 kB | 145 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 626 B | 626 B | ✓ |
| middleware-r..fest.js gzip | 150 B | 151 B | N/A |
| middleware.js gzip | 35.7 kB | 35.7 kB | N/A |
| edge-runtime..pack.js gzip | 1.92 kB | 1.92 kB | ✓ |
| Overall change | 2.55 kB | 2.55 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | simonhaenisch/next.js patch-2 | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 168 kB | 168 kB | ✓ |
| app-page-exp..prod.js gzip | 93.8 kB | 93.8 kB | ✓ |
| app-page-tur..prod.js gzip | 94.5 kB | 94.5 kB | ✓ |
| app-page-tur..prod.js gzip | 89.1 kB | 89.1 kB | ✓ |
| app-page.run...dev.js gzip | 138 kB | 138 kB | ✓ |
| app-page.run..prod.js gzip | 88.4 kB | 88.4 kB | ✓ |
| app-route-ex...dev.js gzip | 24.2 kB | 24.2 kB | ✓ |
| app-route-ex..prod.js gzip | 16.8 kB | 16.8 kB | ✓ |
| app-route-tu..prod.js gzip | 16.9 kB | 16.9 kB | ✓ |
| app-route-tu..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| app-route.ru...dev.js gzip | 23.6 kB | 23.6 kB | ✓ |
| app-route.ru..prod.js gzip | 16.4 kB | 16.4 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.64 kB | 9.64 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.37 kB | 9.37 kB | ✓ |
| pages-turbo...prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| pages.runtim...dev.js gzip | 22.6 kB | 22.6 kB | ✓ |
| pages.runtim..prod.js gzip | 21.9 kB | 21.9 kB | ✓ |
| server.runti..prod.js gzip | 49.3 kB | 49.3 kB | ✓ |
| Overall change | 930 kB | 930 kB | ✓ |

</details>

</details>
Commit: 13fd666d7412ec741687ca254e6674bb72b71b35

**Pull request merged**